attest/ake/src/initiator.rs

```diff
@@ -212,19 +210,90 @@ where
         self,
         _csprng: &mut R,
         input: UnverifiedAttestationEvidence,
-    ) -> Result<(Terminated, VerificationReport), Self::Error> {
+    ) -> Result<(Terminated, EvidenceKind), Self::Error> {
         let output = self
             .state
             .read_message(input.as_ref())
             .map_err(Error::HandshakeRead)?;
         match output.status {
             HandshakeStatus::InProgress(_state) => Err(Error::HandshakeNotComplete),
             HandshakeStatus::Complete(_) => {
-                let remote_report = VerificationReport::decode(output.payload.as_slice())
-                    .map_err(|_e| Error::AttestationEvidenceDeserialization)?;
+                if let Ok(remote_evidence) = EvidenceKind::from_bytes(output.payload.as_slice()) {
+                    Ok((Terminated, remote_evidence))
+                } else {
+                    let remote_report = VerificationReport::decode(output.payload.as_slice())
+                        .map_err(|_| Error::AttestationEvidenceDeserialization)?;
 
-                Ok((Terminated, remote_report))
+                    Ok((Terminated, EvidenceKind::Epid(remote_report)))
+                }
```

Comment:
I think this can be reduced to
```suggestion
                let remote_evidence = EvidenceKind::from_bytes(output.payload.as_slice())
                    .map_err(|_| Error::AttestationEvidenceDeserialization)?;
                Ok((Terminated, remote_evidence))
```